pkg/search/postgres/query/uuid_query.go

```diff
@@ -0,0 +1,108 @@
+package pgsearch
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+func newUUIDQuery(ctx *queryAndFieldContext) (*QueryEntry, error) {
+	whereClause, err := newUUIDQueryWhereClause(ctx.qualifiedColumnName, ctx.value, ctx.queryModifiers...)
+	if err != nil {
+		return nil, err
+	}
+	return qeWithSelectFieldIfNeeded(ctx, &whereClause, nil), nil
+}
+
+func newUUIDQueryWhereClause(columnName string, value string, queryModifiers ...pkgSearch.QueryModifier) (WhereClause, error) {
+	if len(value) == 0 {
+		return WhereClause{}, errors.New("value in search query cannot be empty")
+	}
+
+	if value == pkgSearch.WildcardString {
+		return WhereClause{
+			Query:  fmt.Sprintf("%s is not null", columnName),
+			Values: []interface{}{},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				foundVal := strings.ToLower(foundValue.(string))
+				return foundVal != ""
+			},
+		}, nil
+	}
+
+	if len(queryModifiers) == 0 {
+		uuidVal, err := uuid.FromString(value)
+		if err != nil {
+			return WhereClause{}, errors.Wrapf(err, "value %q in search query must be valid UUID", value)
+		}
+		return WhereClause{
+			Query:  fmt.Sprintf("%s = $$", columnName),
+			Values: []interface{}{uuidVal},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				return strings.EqualFold(foundValue.(string), value)
+			},
+		}, nil
+	}
+	if queryModifiers[0] == pkgSearch.AtLeastOne {
+		panic("I dont think this is used")
+	}
+	var negationString string
+	negated := queryModifiers[0] == pkgSearch.Negation
+	if negated {
+		negationString = "!"
+		if len(queryModifiers) == 1 {
+			uuidVal, err := uuid.FromString(value)
+			if err != nil {
+				return WhereClause{}, errors.Wrapf(err, "value %q in search query must be valid UUID", value)
+			}
+			return WhereClause{
+				Query:  fmt.Sprintf("%s != $$", columnName),
+				Values: []interface{}{uuidVal},
+				equivalentGoFunc: func(foundValue interface{}) bool {
+					return strings.EqualFold(foundValue.(string), value) != negated
+				},
+			}, nil
+		}
+		queryModifiers = queryModifiers[1:]
+	}
+
+	switch queryModifiers[0] {
+	case pkgSearch.Regex:
+		re, err := regexp.Compile(value)
+		if err != nil {
+			return WhereClause{}, fmt.Errorf("invalid regexp %s: %w", value, err)
+		}
+
+		if value != "*" {
```

Comment:
I might be missing some context but why only `*` is supported and not other regexes, for example, `abc.*`?

Reply:
@md2119 I didn't think that made a lot of sense for a UUID.  What would be the use case for that since UUIDs are random?  There is no logical grouping of say all UUIDs that start with `abc*`.  If we need to cover that, I can certainly add it.

Reply:
This is a breaking change given that we currently support any regex on the string fields which are planned to be converted uuid. At the very least we should call it out in the docs.

Reply:
@md2119 got it.  Probably easy enough to just support.  I will take a look at that capability in with PR https://github.com/stackrox/stackrox/pull/3698